client/src/context/SessionContext.js

```diff
@@ -0,0 +1,37 @@
+import { createContext, useEffect } from 'react'
+import Cookies from 'js-cookie'
+import { HOST } from '../util/apiRoutes'
+
+export const SessionContext = createContext()
+
+export const SessionContextProvider = ({ children }) => {
+    useEffect(() => {
+        // Function to initialize session
+        const initializeSession = async () => {
+            const currentSessionId = Cookies.get('sessionId')
+            console.log(currentSessionId)
+
+            if (!currentSessionId || currentSessionId === 'undefined') {
+                // Use fetch to get a new session ID if one does not already exist
+                try {
+                    const response = await fetch(`${HOST}/api/token`)
+
+                    if (!response.ok) {
+                        throw new Error('Network response was not ok')
```

Comment:
what we want to do here is check if the session is expired. so if the token that we used before is too old, then we need to regenerate a token and send it here to be used.